src/pkg/account/m365.go

```diff
@@ -57,6 +62,31 @@ func (a Account) M365Config() (M365Config, error) {
 	return c, c.validate()
 }
 
+func (a Account) GetM365ConfigForHashing() (map[string]any, error) {
```

Comment:
Why not just introduce something like `Hash() (string, error)` on repository?

Reply:
@meain This is to get the part of M365Config (for exclusion of fields that we do not want to be cosidered for hash) for hash generation.
Simlarly for S3: https://github.com/alcionai/corso/blob/43fc461138842b9d2f431aa9c1ce3ed665722b3e/src/pkg/storage/s3.go#L78

, and Filesystem: https://github.com/alcionai/corso/blob/43fc461138842b9d2f431aa9c1ce3ed665722b3e/src/pkg/storage/filesystem.go#L39

In repository, we have: https://github.com/alcionai/corso/blob/43fc461138842b9d2f431aa9c1ce3ed665722b3e/src/pkg/repository/repository.go#L286